MetaMorpheus/EngineLayer/PeptideSpectralMatch.cs

```diff
@@ -499,5 +510,66 @@ protected PeptideSpectralMatch(PeptideSpectralMatch psm, List<(int Notch, Peptid
             SpectralAngle = psm.SpectralAngle;
         }
 
+        public int Compare(object x, object y)
+        {
+            return Compare((PeptideSpectralMatch)x, (PeptideSpectralMatch)y);
```

Comment:
Does this call the CompareTo method under the hood? What happens if one or more casts fail?